.github/workflows/publish-crates.yml

```diff
@@ -0,0 +1,80 @@
+name: Publish Crates
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish (without v prefix)'
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  publish-crates:
+    name: Publish crates to crates.io
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Update versions in Cargo.toml files
+      - name: Update crate versions
+        run: |
+          VERSION=${{ inputs.version }}
+          # Update version in each Cargo.toml
+          for crate in common crypto serde class-hash; do
+            # Update the package version
+            sed -i "s/^version = \".*\"/version = \"$VERSION\"/" "crates/$crate/Cargo.toml"
+            # Also update any internal dependencies to use the same version
+            sed -i "s/{ path = \"..\/common\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+            sed -i "s/{ path = \"..\/crypto\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+            sed -i "s/{ path = \"..\/serde\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+          done
```

Comment:
I'm not sure this is good practice. In general, this would lead to the published versions to be different compared to what we have in git.

Can't we just verify this instead of changing stuff in Cargo.toml files? Although by the time this runs it's already fairly late, we should just make sure that the release action fails completely in case publishing the crates fails.

Reply:
Tracking this in a new issue: #2509